example_configs/cassandra.yml

```diff
@@ -1,14 +1,58 @@
----
 lowercaseOutputLabelNames: true
 lowercaseOutputName: true
+whitelistObjectNames: ["org.apache.cassandra.metrics:*"]
+# ColumnFamily is an alias for Table metrics
+blacklistObjectNames: ["org.apache.cassandra.metrics:type=ColumnFamily,*"]
 rules:
-- pattern: org.apache.cassandra.metrics<type=(Connection|Streaming), scope=(\S*), name=(\S*)><>(Count|Value)
-  name: cassandra_$1_$3
+# Generic gauges with 0-2 labels
+- pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=(\S*)><>Value
+  name: cassandra_$1_$5
+  type: GAUGE
+  labels:
+    "$1": "$4"
+    "$2": "$3"
+
+#
+# Emulate Prometheus 'Summary' metrics for the exported 'Histogram's.
+# TotalLatency is the sum of all latencies since server start
+#
+- pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=(.+)?(?:Total)(Latency)><>Count
+  name: cassandra_$1_$5$6_seconds_sum
+  type: COUNTER
   labels:
-    address: "$2"
-- pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?,
-    name=(\S*)><>(Count|Value)
+    "$1": "$4"
+    "$2": "$3"
+  # Convert microseconds to seconds
+  valueFactor: 0.000001
+
+- pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=((?:.+)?(?:Latency))><>Count
+  name: cassandra_$1_$5_seconds_count
+  type: COUNTER
+  labels:
+    "$1": "$4"
+    "$2": "$3"
+
+- pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=(.+)><>Count
+  name: cassandra_$1_$5_count
```

Comment:
This is picking up counters too, which would be _total usually.

Reply:
I went back and forth on this. There doesn't appear to be a reliable way to determine whether the `<>Count` is part of a DropWizard Histogram/Timer or if it's an individual counter. 

The compromise I came to was choosing the naming scheme of a Summary which matches the majority of the metrics, and it still clearly denotes a counter (despite not matching best-practices for naming of counters with `_total`).
> the count of events that have been observed, exposed as `<basename>_count`

Reply:
Looking through the file, I think you can do that if it has histogram in the name it's a count otherwise total.

Reply:
There are some other Histograms (Timers included) that don't have "histogram" in the name.

Would you prefer that they are defined each of these in the configuration for `_count`? 
We could then set all other remaining matches for `<>Count` to `<basename>_total`
Example:
```
- pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=(\S+Histogram|ViewReadTime|ViewLockAcquireTime|...)><>Count
  name: cassandra_$1_$5_count
  type: COUNTER
  labels:
    "$1": "$4"
    "$2": "$3"

- pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=(.+)><>Count
  name: cassandra_$1_$5_total
  type: COUNTER
  labels:
    "$1": "$4"
    "$2": "$3"
```
Reference: http://cassandra.apache.org/doc/latest/operating/metrics.html

Name | Type | Description
 --- | --- | ---
AnticompactionTime | Timer | Time spent anticompacting before a consistent repair.
BytesValidated | Histogram | Histogram over the amount of bytes read during validation.
Hints_delays | Histogram | Histogram of hint delivery delays (in milliseconds)
Hints_delays-\<PeerIP\> | Histogram | Histogram of hint delivery delays (in milliseconds) per peer
IndexedEntrySize | Histogram | Histogram of the on-heap size, in bytes, of the index across all SSTables.
IndexInfoCount | Histogram | Histogram of the number of on-heap index entries managed across all SSTables.
IndexInfoGets | Histogram | Histogram of the number index seeks performed per SSTable.
PartitionsPerCounterBatch | Histogram | Distribution of the number of partitions processed per counter batch
PartitionsPerLoggedBatch | Histogram | Distribution of the number of partitions processed per logged batch
PartitionsPerUnloggedBatch | Histogram | Distribution of the number of partitions processed per unlogged batch
PartitionsValidated | Histogram | Histogram over the number of partitions read during validation.
RepairPrepareTime | Timer | Total time spent preparing for repair.
RepairTime | Timer | Total time spent as repair coordinator.
SyncTime | Timer | Time spent doing streaming during repair.
ValidationTime | Timer | Time spent doing validation compaction during repair.
ViewLockAcquireTime | Timer | Time taken acquiring a partition lock for materialized view updates on this table.
ViewReadTime | Timer | Time taken during the local read of a materialized view update.
WaitingOnCommit | Timer | The time spent waiting on CL fsync; for Periodic this is only occurs when the sync is lagging its sync interval.
WaitingOnFreeMemtableSpace | Histogram | Histogram of time spent waiting for free memtable space, either on- or off-heap.
WaitingOnSegmentAllocation | Timer | Time spent waiting for a CommitLogSegment to be allocated - under normal conditions this should be zero.

Reply:
It's probably easiest to do leave them all as _count, though several of the metrics in that list aren't in your output.

Reply:
I've noticed that some of the metrics won't always show in the mBeans until they're called, unfortunately. My output was taken from a Cassandra 3.11 server in a development environment. There may be some functions which have not been called since Cassandra had been restarted.

There are some differences from the "latest" (4.0) docs, that I referenced above, compared to 3.11. That explains why some of them are not showing in my output. Sorry for the confusion.

3.11 docs: https://github.com/apache/cassandra/blob/cassandra-3.11/doc/source/operating/metrics.rst

Reply:
Actually thinking about this we should make these _total so that when the java client supports openmetrics it doesn't become _count_total as it's marked as a COUNTER.

Reply:
Would that also apply for the `_sum` metrics as well? Or only for the `_count` metrics?
```
 - pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=((?:.+)?(?:Latency))><>Count
-  name: cassandra_$1_$5_seconds_count
+  name: cassandra_$1_$5_seconds_total
   type: COUNTER

 - pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=(.+)><>Count
-  name: cassandra_$1_$5_count
+  name: cassandra_$1_$5_total
   type: COUNTER
```

Reply:
We could list them as UNTYPED

Reply:
How's this?
```

Reply:
If you're doing this, you can go back to _count.

Reply:
Let's go with UNTYPED so that we'll follow the Summary naming standards and since we cannot guarantee that the metrics under the `<>Count` catch-all is necessarily a Counter.

Does that work for you?
```diff
 - pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=(.+)?(?:Total)(Latency)><>Count
   name: cassandra_$1_$5$6_seconds_sum
-  type: COUNTER
+  type: UNTYPED
   labels:
     "$1": "$4"
     "$2": "$3"

 - pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=((?:.+)?(?:Latency))><>Count
   name: cassandra_$1_$5_seconds_count
-  type: COUNTER
+  type: UNTYPED
   labels:
     "$1": "$4"
     "$2": "$3"

 - pattern: org.apache.cassandra.metrics<type=(\S*)(?:, ((?!scope)\S*)=(\S*))?(?:, scope=(\S*))?, name=(.+)><>Count
   name: cassandra_$1_$5_count
-  type: COUNTER
+  type: UNTYPED
   labels:
```

Reply:
Thanks, @brian-brazil! Updated.